include/bl_storage.h

```diff
@@ -7,15 +7,65 @@
 #ifndef BL_STORAGE_H_
 #define BL_STORAGE_H_
 
-#include <zephyr/types.h>
 #include <string.h>
+#include <zephyr/types.h>
+#include <drivers/nrfx_common.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
 
+#if defined(CONFIG_IS_SECURE_BOOTLOADER) || defined(CONFIG_SECURE_BOOT)
+#include <pm_config.h>
+#define PROVISION_PARTITION_ADDRESS PM_PROVISION_ADDRESS
+#else
+#define PROVISION_PARTITION_ADDRESS CONFIG_SECURE_BOOT_STORAGE_START_ADDRESS
+#endif
+
 #define EHASHFF 113 /* A hash contains too many 0xFs. */
+#define EREADLCS 114 /* LCS field of OTP is in an invalid state */
+#define EINVALIDLCS 115 /* Invalid LCS*/
+
+/** Storage for the PRoT Security Lifecycle state, that consists of 4 states:
+ *  - Device assembly and test
+ *  - PRoT Provisioning
+ *  - Secured
+ *  - Decommissioned
+ *  These states are transitioned top down during the life time of a device.
+ *  Therefore when setting a new LCS we just mark the old state as left, this
```

Comment:
The documentation doesn't correspond to the changes in how we describe the LCS states